### DISCUSSION TITLE
What's the proper way to manually create clips in Playtime?

### DISCUSSION THREAD

**Discussion by on3iro** [helgoboss/helgobox#1116] [Q&A] [answered]
Hey,

yesterday I toyed around with playtime and at one point wanted to simply create a clip by hand instead of playing with a controller.
There where various things that left me a bit confused:

1. I couldn't really figure out what the correct way to adjust the clip length is: I found two parameters for dynamic and static length and they more or less worked. However it is still not clear to me, what the difference is and when I should adjust which parameter. In the end I seemed to need a mix of both to achieve the result. (At one point I saw that the length inside the midi editor was adjusted, but the playhead still stopped playing after one beat.
2. The setting mentioned in 1. says "beat" (in german I would think of a "Takt"), however I seemed to adjust quarter notes instead of a full beat
3. Whenever I tried to adjust the length by dragging the corners of the midi clip in the midi editor instead, the timing of the clip suddenly was really off and even reverting the change didn't fix the issue. 

Could you please explain how I can adjust the length and also how I can create midi clips by hand. (I assume one workaround would be to create the clip in reaper first and then just import it, but that seems to be pretty cumbersome and would disrupt the flow a bit)

Thanks in advance :)

**Comment by helgoboss** [comment, marked as answer]
> 1. I couldn't really figure out what the correct way to adjust the clip length is: I found two parameters for dynamic and static length and they more or less worked. However it is still not clear to me, what the difference is and when I should adjust which parameter. In the end I seemed to need a mix of both to achieve the result. (At one point I saw that the length inside the midi editor was adjusted, but the playhead still stopped playing after one beat.

Imagine you cut some piece out of the paper. The result is the outer section ("Static/fixed section"). Then you cut again something out of outer section. The result is the inner section ("Dynamic section").

The outer/static /fixed section is a section that is not intended to change much. It's mainly used to permanently remove unnecessary material at the beginning or end of the recording. It's primarily important for audio. For MIDI, I personally wouldn't use it because there you have a much better way to permanently change the length and start of the clip: By using the MIDI editor (drag the right edge to the left and you will shorten it, drag it to the right and you will prolong it).

The inner/dynamic section exists so that you can play around with different start points and lengths during a performance - without having to touch the static section.

> 2. The setting mentioned in 1. says "beat" (in german I would think of a "Takt"), however I seemed to adjust quarter notes instead of a full beat

No, "beat" = "Schlag", "bar" = "Takt". And because you probably have a time signature of n/4 (e.g. 4/4), one beat corresponds to one quarter note. So you are adjusting in quarter notes here.

> 3. Whenever I tried to adjust the length by dragging the corners of the midi clip in the midi editor instead, the timing of the clip suddenly was really off and even reverting the change didn't fix the issue.

When you adjust the length during playback by using the MIDI editor, it's quite easy to lose timing. It's only natural. If you want things to stay in time, you must make sure that you adjust the lengths in multiples only. That's quite hard to get right in the MIDI editor during playback.

Once the timing is off, you need to revert the change **and** retrigger clip playback.

For the section controls, Playtime offers convenient "halve" and "double" buttons in order to change the length in multiples only. If the clip is configured to sync to the project tempo, Playtime will automatically retrigger and apply the section change at the next quantization interval (e.g. the next bar). This keeps timing perfect.

### In a nutshell

For audio clips:

- In order to crop the audio recording to the part that you want to play in general, use the static section. Playtime sets this section automatically when recording audio with quantized start and stop timing.
- If you want to play around with the start point and length, use the dynamic section.
- Make sure to use only halving and doubling if you don't want to destroy the timing.

For MIDI clips:

- In order to crop the MIDI recording to the part that you want to play in general, use the MIDI editor. You could also use the static section, but with MIDI that's not necessary.
- If you want to play around with the start point and length, use the dynamic section.
- Make sure to use only halving and doubling if you don't want to destroy the timing.
 
> Could you please explain how I can adjust the length and also how I can create midi clips by hand. (I assume one workaround would be to create the clip in reaper first and then just import it, but that seems to be pretty cumbersome and would disrupt the flow a bit)

1. Double-click the slot
2. Double-click the resulting MIDI clip
3. Drag the right edge as you desire (probably to the multiple of a bar)

BTW, you need to drag the "left" part of the right edge, otherwise you just change what's visible, not the actual length.

**Reply by on3iro** [reply]
@helgoboss Awesome - thanks a ton for the detailed response (again). Your response times are insane 😄  ❤️ 

ps: of course its `beat` and `bar` - my brain somehow totally messed that up 😅

**Reply by helgoboss** [reply]
I just added part of this explanation to the Wiki: https://github.com/helgoboss/helgobox/wiki/Playtime-FAQ ... everyone feel free to improve it / make it more understandable.
